### ISSUE TITLE
Transfer learning ResNet mode performance drop in recent versions

### ISSUE THREAD

**Issue by GCidd** [tensorflow/tensorflow#51790]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
Ubuntu 16.04, 18.04, 20.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
No
- TensorFlow installed from (source or binary):
Binary
- TensorFlow version (use command below):
2.0.0, 2.2.0, 2.4.0, 2.5.0, 2.6.0
- Python version:
3.7, 3.8, 3.9
- Bazel version (if compiling from source):
No
- GCC/Compiler version (if compiling from source):
No
- CUDA/cuDNN version:
10.2, 11.1
- GPU model and memory:
1660 Super (6GB), 3070 Laptop (8GB), 3090 (24GB), whatever google's colab provides

**Describe the current behavior**
ResNet (50, 101 and 152) perform at about 40-50% Accuracy and F1 score after transfer learning.
**Describe the expected behavior**
6-8 months ago, using the same exactly dataset and script, the same models performed above 75% at the same metrics.
The initial good results were obtained using google colab.
Now, I have tested the **same code and dataset** in google colab using all the tensorflow versions mentioned above.
Have also tested on Ubuntu 16.04 with CUDA 10.2 and CUDNN 7.6 and Ubuntu 18.04 and 20.04 with CUDA 11.1 CUDNN 8.2.

The same results (old and new) are consistent (+- 1%) for all the other models I have used (All DenseNets, VGG16 and 19 and MobileNet).

**[Contributing](https://www.tensorflow.org/community/contribute)**

- Do you want to contribute a PR? (yes/no):
No
- Briefly describe your candidate solution(if contributing):
Not sure

**Standalone code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate
the problem. If possible, please share a link to Colab/Jupyter/any notebook.
Dataset is private so I can't.

**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.
Transfer learning process follows the one provided [by keras](https://keras.io/guides/transfer_learning/).

**Comment by sushreebarsa** [comment]
@GCidd Please post this issue on [keras-team/keras repo.](https://github.com/keras-team/keras/issues)
To know more see;
[https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999](https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999)
Thank you!
